AzureStack Network Admin Specification

deathly809 opened the pull request:
New specification for AzureStack Network admin endpoints.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Commit:
AzureStack Network Admin Specification

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-libraries-for-java

AutorestCI commented:
# Automation for azure-sdk-for-node

anuchandy commented on specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/2015-06-15/Network.json:
```diff
@@ -0,0 +1,283 @@
+                    "description": "The tenant resource URI.",
+                    "type": "string"
+                }
+            }
```
Are there any properties of the `ResourceTenant` that can be marked as readonly ? seems provisioningState can be readonly right?

anuchandy commented on specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/2015-06-15/Network.json:
```diff
@@ -0,0 +1,283 @@
+                }
+            }
+        }
+    },
```
can you also check whether properties and nested properties of `AdminOverview` model are eligible to be marked as readonly?

anuchandy commented on specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/2015-06-15/Quotas.json:
```diff
@@ -0,0 +1,310 @@
+                        "schema": {
+                            "$ref": "#/definitions/Quota"
+                        }
+                    }
```
If this is a long running operation then please mark so e.g. https://github.com/Azure/azure-rest-api-specs/blob/a93922b09f8ccba9899f8dc8c3cc9de6e5b76776/specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json#L533

deathly809 replied:
It isn't.  Long running operations should return 202 and then eventually 200/201 correct?

anuchandy commented on specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/2015-06-15/VirtualNetworks.json:
```diff
@@ -0,0 +1,207 @@
+                        "$ref": "#/definitions/VirtualNetworkConfigurationStatus"
+                    }
+                }
+            }
```
any of the properties can be marked as readonly here?

anuchandy commented:
@ravbhatnagar new swagger. Please review from ARM side.

anuchandy commented on specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/2015-06-15/Network.json:
```diff
@@ -0,0 +1,283 @@
+                "location": {
+                    "description": "Region location of resource.",
+                    "type": "string",
+                    "readOnly": true
```
Resource::location is readonly, Quote inherits from Resource, so does this means user cannot specify the location on which the Quote he want to create?

deathly809 replied:
The quota endpoints use location based routing, the location property is not used except when set by the RP.

deathly809 commented:
@anuchandy We sent out a code review to the Network team and they already approved the changes.  What is in the spec is what they said was correct.

anuchandy reviewed (changes requested):
Please see the comments

Commit:
Moved to preview

Commit:
Fixed readonly issues and updated to preview

deathly809 commented:
@anuchandy I have updated the specs and put readonly in places where they should be.  I have updated to be preview as well.  This is not going in Azure FYI.

anuchandy commented:
@deathly809 thank you, marking them as readonly ensure user won't get un-necessary setters.

anuchandy commented:
@ravbhatnagar Ok, thanks for letting me know.
